### ISSUE TITLE
library of Node.js facilities

### ISSUE THREAD

**Issue by dherman** [mozilla/task.js#14]
Just like `task.dom.js` there should be a `task.node.js` file that wraps standard Node libraries with promise abstractions for working with task.js. These won't really be useable until there's a version of task.js that implements generators, but once V8 implements generators, Node's async I/O is another good use case for task.js.

(Depends on issue #12.)

**Comment by dherman** [comment]
Out of scope for task.js; these can simply be done as some node promises library, which would automatically be compatible with task.js.
